packages/connect-disputable-voting/src/models/Vote.ts

```diff
@@ -75,23 +81,35 @@ export default class Vote {
     this.quietEndingSnapshotSupport = data.quietEndingSnapshotSupport
     this.script = data.script
     this.settledAt = data.settledAt
+    this.disputedAt = data.disputedAt
     this.executedAt = data.executedAt
     this.isAccepted = data.isAccepted
     this.tokenDecimals = data.tokenDecimals
+    this.submitterArbitratorFeeId = data.submitterArbitratorFeeId
+    this.challengerArbitratorFeeId = data.challengerArbitratorFeeId
   }
 
   get hasEnded(): boolean {
-    return this.voteStatus === 'Cancelled' || (
+    return this.voteStatus === 'Cancelled' || this.voteStatus === 'Settled' || (
       this.voteStatus !== 'Challenged' &&
       this.voteStatus !== 'Disputed' &&
       Date.now() >= toMilliseconds(this.endDate)
     )
   }
 
   get endDate(): string {
-    const originalEndDate = bn(this.startDate).add(bn(this.duration))
-    const endDateAfterPause = originalEndDate.add(bn(this.pauseDuration))
-    return endDateAfterPause.add(bn(this.quietEndingExtensionDuration)).toString()
+    const baseVoteEndDate = bn(this.startDate).add(bn(this.duration))
+    const endDateAfterPause = baseVoteEndDate.add(bn(this.pauseDuration))
+    const lastComputedEndDate = endDateAfterPause.add(bn(this.quietEndingExtensionDuration))
+
+    // The last computed end date is correct if we have not passed it yet or if no flip was detected in the last extension
+    const currentTimestamp = bn(parseInt((Date.now() / 1000).toString()).toString())
```

Comment:
No need to change anything, but I was wondering if there was reason to do `toString()` => `parseInt()` => `toString()`? Or if we could use `String(Math.floor(Date.now() / 1000))`?

Reply:
Just a brainfart haha, we can do that for sure :)